File: j-LIB-core/src/main/java/io/github/dot166/jLib/utils/NetUtils.java
package io.github.dot166.jLib.utils;

import android.content.Context;
import android.database.Cursor;
import android.net.Uri;
import android.os.StrictMode;
import android.util.Log;
import android.widget.Toast;

import androidx.annotation.NonNull;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;
import java.util.HashMap;

public class NetUtils {

    public static String getDataRaw(String urlString, @NonNull Context context) {
        StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitNetwork().build();
        StrictMode.setThreadPolicy(policy);
        try {
            URL url = new URL(urlString);
            HttpURLConnection connection = ((HttpURLConnection) url.openConnection());
            InputStream in_stream = connection.getInputStream();
            return inputSteramToString(in_stream);
        } catch (Exception e) {
            ErrorUtils.handle(e, context);
            return "";
        }
    }

    public static String inputSteramToString(InputStream in) throws IOException {
        StringBuilder builder = new StringBuilder();
        BufferedReader reader = new BufferedReader(new InputStreamReader(in));
        String line;
        while ((line = reader.readLine()) != null) {
            builder.append(line).append("\n");
        }
        reader.close();
        return builder.toString();
    }
}

File: j-LIB-core/src/main/java/io/github/dot166/jlib/utils/NetUtils.kt
package io.github.dot166.jlib.utils

import android.content.Context
import android.os.StrictMode
import android.os.StrictMode.ThreadPolicy
import io.github.dot166.jlib.utils.ErrorUtils.handle
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.URL

object NetUtils {
    fun getDataRaw(urlString: String?, context: Context, timeoutMs: Int = 2000): String {
        val policy = ThreadPolicy.Builder().permitNetwork().build()
        StrictMode.setThreadPolicy(policy)
        try {
            val url = URL(urlString)
            val connection = (url.openConnection() as HttpURLConnection)
            connection.setConnectTimeout(timeoutMs)
            return inputSteramToString(connection.getInputStream())
        } catch (e: Exception) {
            handle(e, context)
            return ""
        }
    }

    @Throws(IOException::class)
    fun inputSteramToString(`in`: InputStream?): String {
        val builder = StringBuilder()
        val reader = BufferedReader(InputStreamReader(`in`))
        var line: String?
        while ((reader.readLine().also { line = it }) != null) {
            builder.append(line).append("\n")
        }
        reader.close()
        return builder.toString()
    }
}
